### PR TITLE
volgen: Fix bitrot volgen issue when no local bricks in a subvolume

### PR DESCRIPTION
**Pull request by aravindavk**
Updates: #815
Signed-off-by: Aravinda VK <avishwan@redhat.com>

### CODE REVIEW

File: glusterd2/volgen2/clustergraph.go [hunk 1]
```diff
@@ -46,6 +46,23 @@ func clusterGraph(volfile *Volfile, dht *Entry, vol *volume.Volinfo, nodeid uuid
 			}
 		}
 
+		// If Graph is required only for Local Bricks then
```
Comment (Madhu-1) [inline glusterd2/volgen2/clustergraph.go]: to be cleaner can we use continue with LABEL like below 

```
LABEL:
if filters != nil && filters.onlyLocalBricks {
			
	for _, b := range subvol.Bricks {
			if uuid.Equal(b.PeerID, nodeid) {
				continue LABEL
			}
         }
}
```

Comment (aravindavk) [reply glusterd2/volgen2/clustergraph.go]: nice feature, but in this case not much useful. Your example will go wrong and loop will be skipped if one local brick found

Comment (Madhu-1) [reply glusterd2/volgen2/clustergraph.go]: @aravindavk agreed

**Comment by prashanthpai** [comment]
retest this please

File: glusterd2/volgen2/clustergraph.go [hunk 2]
```diff
@@ -46,6 +46,23 @@ func clusterGraph(volfile *Volfile, dht *Entry, vol *volume.Volinfo, nodeid uuid
...
+		// do not include sub volume in graph if no local bricks
+		// exists for the sub volume
+		if filters != nil && filters.onlyLocalBricks {
+			localBricksExists := false
+			for _, b := range subvol.Bricks {
```
[the remark this answers is not in the document]
Comment (aravindavk) [reply glusterd2/volgen2/clustergraph.go]: > Why not add a method on Subvol to get local bricks. Like you have it for Volinfo. That would simplify this to,

Nice, I will add the method.

> I saw a similar comment by you on another PR, so just wondering why you didn't think of it. 

that comment is for the existing method :) similar loop code is used in this file so didn't think about it. Thanks, I will add the method to subvol

[the remark this answers is not in the document]
Comment (aravindavk) [reply glusterd2/volgen2/clustergraph.go]: but performance wise loop is better(over optimization) here, exits early once it finds atleast one local brick. :)

**Pull request merged**